lib/langchain/tool/base.rb

```diff
@@ -7,15 +7,15 @@ class Base
     # How to add additional Tools?
     # 1. Create a new file in lib/tool/your_tool_name.rb
     # 2. Add your tool to the TOOLS hash below
-    #   "your_tool_name" => "Tool::YourToolName"
+    #    "Tool::YourToolName" => "your_tool_name"
     # 3. Implement `self.execute(input:)` method in your tool class
     # 4. Add your tool to the README.md
 
     TOOLS = {
-      "calculator" => "Langchain::Tool::Calculator",
-      "search" => "Langchain::Tool::SerpApi",
-      "wikipedia" => "Langchain::Tool::Wikipedia",
-      "database" => "Langchain::Tool::Database"
+      "Langchain::Tool::Calculator" => "calculator",
+      "Langchain::Tool::SerpApi" => "search",
+      "Langchain::Tool::Wikipedia" => "wikipedia",
+      "Langchain::Tool::Database" => "database"
```

Comment:
What if instead of this hash map, we just had a `name` or `tool_name` method on the Tool class itself? For example you can call the `DESCRIPTION` constant on each tool:
```ruby
Langchain::Tool::Calculator::DESCRIPTION
```

Reply:
I thought maybe this was a safety mechanism to avoid passing in any 'unregistered' tool_name.  See below.